docs/exts/docroles.py

```diff
@@ -0,0 +1,82 @@
+#
```

Comment:
Yeah. I think we should separate those out :). Maybe we should make a separate commit with those new classes and make it a prerequisite for this one?

Reply:
If we split everything, we will not be able to introduce changes. We have a lot of open changes and we have to close them. If the community did report it, it will split.